node_modules/oae-principals/lib/api.user.js

```diff
@@ -130,52 +133,69 @@ var createUser = module.exports.createUser = function(ctx, tenantAlias, displayN
     validator.check(opts.visibility, {'code': 400, 'msg': 'The specified visibility setting is unknown'}).isIn(_.values(AuthzConstants.visibility));
     validator.check(opts.emailPreference, {'code': 400, 'msg': 'The specified email preference is invalid'}).isIn(_.values(PrincipalsConstants.emailPreferences));
 
+    // Because some SSO strategies do not release an email address, we allow user accounts to be
+    // created without providing the email.
+    // TODO: The UI however, will keep nagging users about filling one in
+    if (opts.email) {
+        validator.check(opts.email, {'code': 400, 'msg': 'The specified email address is invalid'}).isEmail();
+    }
     if (validator.hasErrors()) {
         return callback(validator.getFirstError());
     }
 
-    var id = AuthzUtil.toId('u', tenantAlias, ShortId.generate());
-
-    var values = {
-        'tenantAlias': tenantAlias,
-        'displayName': displayName,
-        'visibility': opts.visibility,
-        'email': opts.email,
-        'emailPreference': opts.emailPreference,
-        'locale': opts.locale,
-        'publicAlias': opts.publicAlias,
-        'smallPictureUri': opts.smallPictureUri,
-        'mediumPictureUri': opts.mediumPictureUri,
-        'largePictureUri': opts.lagePictureUri
-    };
+    // Verify this email address, if any, is not already mapped to another user
+    _getUserByEmailIfNecessary(opts.email, null, function(err, existingUser) {
+        if (err && err.code !== 404) {
+            return callback(err);
+        } else if (existingUser) {
+            return callback({'code': 400, 'msg': 'This email address is already associated with another user'});
```

Comment:
We aren't supposed to be restricting this anymore. You can create as many accounts as you like with the same email.